### ISSUE TITLE
"Exclude" for whole department

### ISSUE THREAD

**Issue by dmke** [rubocop/rubocop#9889]
## Is your feature request related to a problem? Please describe.

I've got a few legacy Rails applications where we're transitioning from MiniTest to RSpec, consequently I'm using `rubocop-rspec` and `rubocop-minitest`. When running `rubocop test/`, I'm getting warnings from `rubocop-rspec`:

```console
$ bundle exec rubocop test/
...
test/unit/user_test.rb:50:9: C: [Correctable] RSpec/Rails/AvoidSetupHook: Use before instead of setup.
        setup do ...
        ^^^^^^^^
...
```

I don't want change `setup` to `before` here. I could disable the cop in my `rubocop.yml`, like so:

```yaml
inherit_from:
- https://company.com/shared/rubocop.yml
- https://company.com/shared/rubocop-rails.yml
- https://company.com/shared/rubocop-rspec.yml

RSpec/Rails/AvoidSetupHook:
  Exclude:
    - test/**/*.rb
```

**However,** `RSpec/Rails/AvoidSetupHook` is just one example. For various reasons, I found the need to disable whole departments for specific directories (another example is a really old legacy library in dire need of refactoring, but for now I don't want Rubocop to touch it in any way) - and excluding one directory in dozens of cops gets tedious.

## Describe the solution you'd like

I'd like to `Exclude` on whole departments:

```yaml
RSpec/*:
  Exclude:
    - test/**/*.rb

Style/*:
  Exclude:
    - vendor/legacylib/**/*.rb
```

I'm not hung-up on the syntax used in the example above (`<Department>/*`). 

## Describe alternatives you've considered

The only alternative I've found was to exclude the directories from each cop, one-by-one. In Rubocop 1.17, there are 215 style cops in total (though, not all enabled by default).

## Additional context

I want to run Rubocop both as pre-commit hook (on changed files), as well as on our CI, however the amount of noise produced in the examples described above (or the configuration work needed to keep that noise level down) prevents me from doing so.

**Comment by athix** [comment]
Also running into this where I need to disable a department for a specific folder.

**Comment by bbatsov** [comment]
I'll leave to @jonas054 to comment of the feasibility of this. The code around inclusions and exclusions is already so complex that I'm always hesitant to add more complexity there.

**Comment by jonas054** [comment]
This is supposed to work already with the syntax
```yaml
RSpec:
  Exclude:
    - test/**/*.rb

Style:
  Exclude:
    - vendor/legacylib/**/*.rb
```
I just tried it on RuboCop's own source code, disabling the `RSpec` department for the `spec` directory tree, removing all other `RSpec` configuration in `.rubocop.yml`, and that worked for me.

Please check if this syntax works for you @dmke and @athix, and report any relevant details if it doesn't.

**Comment by dmke** [comment]
@jonas054: There seems to be some interference with inherited rules:

```yaml
# .ruboco-rspec.yml
require:
  - rubocop-rspec

RSpec/Rails/AvoidSetupHook:
  Enabled: true
```

```yaml
# .rubocop.yml
inherit_from:
  - ./.rubocop-rspec.yml

require:
  - rspec-minitest

AllCops:
  NewCops: enable

RSpec:
  Exclude:
    - test/**/*_test.rb
```

```console
$ bundle exec rubocop                       
Inspecting 2 files
.C

Offenses:

test/unit/user_test.rb:4:3: C: [Correctable] RSpec/Rails/AvoidSetupHook: Use before instead of setup.
  setup do ...
  ^^^^^^^^
```

I've placed a reproduction repo here: https://github.com/dmke/rubocop-9889

*(N.B.: the `setup`/`should` syntax is part of [`shoulda-context`](https://github.com/thoughtbot/shoulda-context))*

**Comment by jonas054** [comment]
My bad. The department for this cop is `RSpec/Rails`, not `RSpec`. So it should be:
```yaml
RSpec/Rails:
  Exclude:
    - test/**/*_test.rb
```

**Comment by dmke** [comment]
That indeed does what I want.

Thank you!
